tests/kafkatest/tests/streams/streams_broker_down_resilience_test.py

```diff
@@ -0,0 +1,64 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from ducktape.tests.test import Test
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.streams import StreamsBrokerDownResilienceService
+from kafkatest.services.zookeeper import ZookeeperService
+
+
+class StreamsBrokerDownResilience(Test):
+    """
+    This test validates that Streams is resilient to a broker
+    being down longer than specified timeouts in configs
+    """
+
+    inputTopic = "streamsResilienceSource"
+    outputTopic = "streamsResilienceSink"
+
+    def __init__(self, test_context):
+        super(StreamsBrokerDownResilience, self).__init__(test_context=test_context)
+        self.zk = ZookeeperService(test_context, num_nodes=1)
+        self.kafka = KafkaService(test_context,
+                                  num_nodes=1,
+                                  zk=self.zk,
+                                  topics={
+                                      self.inputTopic: {'partitions': 1, 'replication-factor': 1},
+                                      self.outputTopic: {'partitions': 1, 'replication-factor': 1}
+                                  })
+
+    def setUp(self):
+        self.zk.start()
+
+    def test_streams_resilient_to_broker_down(self):
+        self.kafka.start()
+
+        processor = StreamsBrokerDownResilienceService(self.test_context, self.kafka, 70000)
+        processor.start()
+
+        node = self.kafka.leader(self.inputTopic)
+
+        self.kafka.stop_node(node)
```

Comment:
Should we make sure that stream is up and running before stopping the broker? we should use python producer and verifyable consumer here to copy a single record from source to sink topic? (Just an idea).

Having said this? If we do this test twice, ie, before broker stop and after broker resume, we can simplify the Java code and get rid of the sleep and verification code and just register an shutdown hook. Might be better test design.